### PR TITLE
Social: Fix isPluginActive checks while activating license

### PR DESCRIPTION
**Pull request by spsiddarthan**
## Proposed changes:

* On JN Beta, the plugin file path is `jetpack-dev/jetpack.php` and `jetpack-social-dev/jetpack-social.php`. On local for social, it's `social/jetpack-social.php` and on self-hosted sites such as on pressable, it's `jetpack-social/jetpack-social.php`. 

This PR incorporates these file name while checking if the Social or the Jetpack plugin is active. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1706712466731389/1706702406.089029-slack-C02JJ910CNL
## Testing instructions:

I have checked this for all the 3 scenarios: Just Jetpack active, just Social active and both Jetpack and Social being active, you could test for any of these. 

* Boot this branch on JN
* Buy Jetpack Social Basic or Advanced, by going to https://cloud.jetpack.com/pricing/
* Go to your site's wp-admin/admin.php?page=my-jetpack#/add-license page to activate the license you just bought.
* You should see the configure my site button.

### CODE REVIEW

File: projects/js-packages/licensing/components/activation-screen/utils.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
+const JETPACK_PLUGIN_FILE_NAME = 'jetpack/jetpack.php';
+const JETPACK_PLUGIN_DEV_FILE_NAME = 'jetpack-dev/jetpack.php';
+const JETPACK_SOCIAL_PLUGIN_FILE_NAME = 'jetpack-social/jetpack-social.php';
+const JETPACK_SOCIAL_PLUGIN_DEV_FILE_NAME = 'jetpack-social-dev/jetpack-social.php';
+const JETPACK_SOCIAL_PLUGIN_LOCAL_FILE_NAME = 'social/jetpack-social.php';
```
Comment (spsiddarthan) [inline projects/js-packages/licensing/components/activation-screen/utils.js:5]: I got these values by examining the file names on local, JN, and my pressable site, which would be the equivalent of a hosted site that users would have.

'jetpack/jetpack.php' and 'jetpack-social/jetpack-social.php' is what we would would see on their self- hosted sites.

File: projects/js-packages/licensing/components/activation-screen/utils.js [hunk 2]
```diff
@@ -64,12 +70,36 @@ export function getProductGroup( productId ) {
...
+
 /**
  * Check if a plugin is active.
  *
- * @param {string} plugin -- The name of the plugin such as jetpack/jetpack.php, social/jetpack-social.php, backup/jetpack-backup.php.
+ * @param {string} plugin -- Plugin filename.
  * @returns {boolean} -- True or False.
  */
-export function isPluginActive( plugin ) {
+function isPluginActive( plugin ) {
```
Comment (pablinos) [inline projects/js-packages/licensing/components/activation-screen/utils.js:103]: If we're in the My Jetpack context (and it correctly identifies that the development versions of plugins are active), I think we might not need this. There is a `useProduct` hook which should give us [information about the product](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx#L28-L35), including whether it's active.

**Review by pablinos — COMMENTED** [review]
This looks reasonable, but we might be able to use the My Jetpack `useProduct` hook to work out if a product is active.

**Comment by spsiddarthan** [comment]
Closing in favour of https://github.com/Automattic/jetpack/pull/35391

**Pull request closed without merging**